lib/internal/modules/cjs/loader.js

```diff
@@ -567,7 +569,16 @@ function tryModuleLoad(module, filename) {
 }
 
 Module._resolveFilename = function(request, parent, isMain, options) {
```

Comment:
This could make this a semver-major. As @mcollina suggests, we need to determine if this breaks anyone. Likely candidates are APM or testing libraries.

Reply:
Maybe instead of modifying `Module._resolveFilename`, we could use a new internal function at the call sites, which detects `@nodejs/` first before calling `Module._resolveFilename` - that would still be a behavior change and may still be semver-major but it would not break `Module._resolveFilename` itself and help us deprecate `Module._resolveFilename` (if it's ever going to be depreated) in the future.

Reply:
On a second thought - maybe the correct place to insert this logic is `NativeModule.nonInternalExists` instead of `Module._resolveFilename`, also `NativeModule.nonInternalExists` is even less public anyways (you'd need `--expose-internals` to get access to `NativeModule`)